internal/model/log.go

```diff
@@ -0,0 +1,38 @@
+package model
+
+// At this limit, with one resource having a 120k log, render time was ~20ms and CPU usage was ~70% on an MBP.
+// 70% still isn't great when tilt doesn't really have any necessary work to do, but at least it's usable.
+// A render time of ~40ms was about when the interface started being noticeably laggy to me.
+const maxLogLengthInBytes = 120 * 1000
+
+type Log struct {
+	content []byte
+}
+
+func NewLog(s string) Log {
+	return Log{[]byte(s)}
+}
+
+func (l Log) String() string {
+	return string(l.content)
+}
+
+func (l Log) Empty() bool {
+	return len(l.content) == 0
+}
+
+func (l Log) Copy() Log {
+	return Log{append([]byte{}, l.content...)}
+}
+
+func (l *Log) Append(b []byte) {
```

Comment:
maybe i'm misunderstanding this func? what happens if the log is:
```shortline\n
really really really really reaaaaaaallly longline\n
more recent stuff...```

even if appending `more recent stuff` gets you MANY bytes over the limit, I think this func will only chop off the very first line (i.e. `shortline`)?

Reply:
I think you're misunderstanding this func?

Does this help?
https://play.golang.org/p/vguuX3pJJYX

I'm kinda not sure how the situation you're describing is unusual enough to be worth commenting, so I worry I don't understand your concern.